Trouble using with TS Node16 moduleResolution

When using TypeScript with Node16 moduleResolution, I get the following error trying to import @apify/storage-local:

```
src/commands/init.ts:2:35 - error TS7016: Could not find a declaration file for module '@apify/storage-local'. 'E:/work/project123/node_modules/.pnpm/@apify+storage-local@2.3.0/node_modules/@apify/storage-local/dist/index.mjs' implicitly has an 'any' type.
  There are types at 'E:/work/project123/node_modules/@apify/storage-local/dist/index.d.ts', but this result could not be resolved when respecting package.json "exports". The '@apify/storage-local' library may need to update its package.json or typings.

2 import { ApifyStorageLocal } from '@apify/storage-local'
```

I found a pertaining issue at https://github.com/microsoft/TypeScript/issues/52363.

The problem boils down to missing an `index.d.mts` file in the dist folder of @apify/storage-local. I made an exact duplicate of the index.d.ts file named index.d.mts and the error went away.

I'm not sure how best to fix this or I'd make a PR. I guess an extremely simple solution would be to modify the postbuild script in package.json to duplicate the index.d.ts file:

```diff
diff --git a/package.json b/package.json
index 6e93985..1e88c0f 100644
--- a/package.json
+++ b/package.json
@@ -41,7 +41,7 @@
     ],
     "scripts": {
         "build": "tsc -b src",
-        "postbuild": "gen-esm-wrapper dist/index.js dist/index.mjs",
+        "postbuild": "gen-esm-wrapper dist/index.js dist/index.mjs && shx cp dist/index.d.ts dist/index.d.mts",
         "prepublishOnly": "npm run build",
         "clean": "tsc -b src --clean",
         "test": "jest",
@@ -74,6 +74,7 @@
         "eslint": "^8.41.0",
         "gen-esm-wrapper": "^1.1.3",
         "jest": "^29.5.0",
+        "shx": "^0.3.4",
         "ts-jest": "^29.1.0",
         "ts-node": "^10.9.1",
         "typescript": "5.1.3"

```

--- a/package.json
+++ b/package.json
@@ -7,13 +7,19 @@
     },
     "main": "dist/index.js",
     "module": "dist/index.mjs",
-    "types": "dist/index.d.ts",
     "exports": {
         ".": {
-            "import": "./dist/index.mjs",
-            "require": "./dist/index.js"
+            "import": {
+                "types": "./dist/index.d.ts",
+                "default": "./dist/index.mjs"
+            },
+            "require": {
+                "types": "./dist/index.d.ts",
+                "default": "./dist/index.js"
+            }
         }
     },
+    "types": "dist/index.d.ts",
     "keywords": [
         "apify",
         "sqlite",

